### PR TITLE
Eval Script

### LINKED ISSUE #647: Eval metrics in DeepLab Tensorboard
It's not clear if the total loss in Tensorboard when using Deeplab is a validation loss or training loss (or if the images in the images tab are from the training or validation set). Also, I was expecting there to be some explicit validation metrics in Tensorboard. (For object detection, there's a validation mAP that shows up). The task is to see what's going on and possibly log this information if possible.

### PR DESCRIPTION
**Pull request by jamesmcclain**
## Overview

Run DeepLab evaluation script upon request.

Test with [this](https://github.com/jamesmcclain/raster-vision-examples/tree/eval) branch.

~~Note: this current logs evaluation events to a directory other than the training log directory.  This is because logging to that directory appears to disrupt the normal logging behavior of the training script (it might be a problem caused by the slow computer that I am using, I will have to investigate more later).~~

The eval script seems to add these:
screenshot_2019-01-07_07-19-48

### Checklist

- [x] Updated `docs/changelog.rst`
- [x] Ran scripts/format_code and committed any changes
- [x] Documentation updated if needed
- [x] PR has a name that won't get you publicly shamed for vagueness

Closes #647

### CODE REVIEW

File: rastervision/backend/tf_deeplab_config.py [hunk 1]
```diff
@@ -316,25 +326,28 @@ def with_train_options(self,
...
            Args:
               sync_interval: How often to sync output of training to the cloud
                 (in seconds).
-
               do_monitoring: Run process to monitor training (eg. Tensorboard)
-
               replace_model: Replace the model checkpoint if exists.
                              If false, this will continue training from
                              checkpoing if exists, if the backend allows for this.
+              do_eval: Boolean determining whether to run the eval
```
[the remark this answers is not in the document]
Comment (jamesmcclain) [reply rastervision/backend/tf_deeplab_config.py]: The eval script is extremely expensive to run.

[the remark this answers is not in the document]
Comment (jamesmcclain) [reply rastervision/backend/tf_deeplab_config.py]: To the best of my knowledge, it uses the same resource(s) as the training script.

**Comment by jamesmcclain** [comment]
Merging shortly...

**Comment by lewfish** [comment]
> Is this a fresh comment? `miou` should show up after some time (after the first checkpoint has been created and the eval script has had time to run).

I tried it a while ago, and that's what happened. Maybe I didn't wait long enough for the first checkpoint.

**Comment by lewfish** [comment]
I'll test it again.

**Comment by lewfish** [comment]
I followed your testing instructions and didn't see the miou panel in Tensorboard even after checkpoints were written.

screen shot 2019-01-18 at 1 45 51 pm

**Comment by jamesmcclain** [comment]
Confirmed working with [this](https://github.com/jamesmcclain/raster-vision-examples/tree/eval) branch.

screenshot_2019-01-21_09-47-29

Keep in mind that this is somewhat timing dependent as well.  I was never able to see the `miou` show up in tensorboard on my laptop because running the training and evaluation scripts at the same time seemed to be too taxing (eval thread seemed to either stop making progress or some other timing bug was exposed).  On a more powerful computer, the `miou` shows up every time.

This goes to my comment above about the eval script being probably too expensive to be practical (or at least desirable).

**Comment by lewfish** [comment]
> Keep in mind that this is somewhat timing dependent as well. I was never able to see the `miou` show up in tensorboard on my laptop because running the training and evaluation scripts at the same time seemed to be too taxing (the eval process seemed to either stop making progress or some other timing bug was exposed). On a more powerful computer, the `miou` shows up every time.
> 
> This goes to my comment above about the eval script being probably too expensive to be practical (or at least desirable).

Ah, ok.

**Comment by lewfish** [comment]
I think this is ready to merge.
